Vagrantfile

```diff
@@ -3,20 +3,22 @@
 
 Vagrant.configure("2") do |config|
   config.vm.box = "ubuntu/bionic64"
-  config.vm.provision "shell", path: "provision.sh"
+  config.vm.provision "shell", path: "provision.bash"
 
   # Don't forward 22.  Even if we do so, the exposed port only binds
   # to 127.0.0.1, but not 0.0.0.0.  Other ports binds to all IPs.
-  config.vm.network "forwarded_port", guest: 3306, host: 3306
-  config.vm.network "forwarded_port", guest: 50051, host: 50051
-  config.vm.network "forwarded_port", guest: 8888, host: 8888
-
-  # public_network eases debugging -- I can SSH from iMac to the VM
-  # without the need of forwarding port to the host workstation.
-  config.vm.network "public_network", ip: "192.168.0.17"
+  config.vm.network "forwarded_port", guest: 3306, host: 3306,
```

Comment:
Does it matter if the host machine already occupied `3306` or `8888`?

Reply:
The parameter `auto_correct:true` would allocate a new port other than 3306/8888 if the port has been taken by some other process.  We can run vagrant ports to see the corrected port forwarding.